### ISSUE TITLE
flaky `DefaultContSpec` in CI [JVM]

### ISSUE THREAD

**Issue by samspills** [typelevel/cats-effect#3594]
On this commit: https://github.com/typelevel/cats-effect/commit/3cc51888d594cf797a9856b7b879d68249d0d642

Failure here: https://github.com/typelevel/cats-effect/actions/runs/4877781790/jobs/8702776117?pr=3593#step:23:4739
for (macos-latest, 2.13.10, graalvm@11, ciJVM)

```
[error] x get within onCancel - 1
[error]  false -> false != true -> true (ContSpec.scala:130)
[error] x get within onCancel - 2
[error]  false -> false != true -> true (ContSpec.scala:151)
```

**Comment by djspiewak** [comment]
Saaaaaaaaadge

**Comment by durban** [comment]
This seems to have been fixed/disappeared (see #3605).
